src/Toolkit/Dom.php

```diff
@@ -51,30 +51,54 @@ public function __construct(
 		$this->doc  = new DOMDocument();
 		$this->type = strtoupper($type);
 
-		// switch to "user error handling"
+		// Switch libxml into internal error handling mode so warnings
+		// don’t leak into output or interrupt parsing
 		$errors = libxml_use_internal_errors(true);
 
 		if ($this->type === 'HTML') {
-			// ensure proper parsing for HTML snippets
+			// If this is an HTML fragment (no <html> or <body> root),
+			// wrap it in <body> so DOMDocument has a valid container.
 			if (preg_match('/<(html|body)[> ]/i', $code) !== 1) {
 				$code = '<body>' . $code . '</body>';
 			}
 
-			// the loadHTML() method expects ISO-8859-1 by default;
-			// force parsing as UTF-8 by injecting an XML declaration
+			// DOMDocument::loadHTML() historically assumes ISO-8859-1 input.
+			// To force UTF-8 parsing, Kirby injects an XML declaration.
+			// The random ID allows us to reliably identify *our* injected node
+			// later and remove it again.
 			$xml  = 'encoding="UTF-8" id="' . Str::random(10) . '"';
 			$load = $this->doc->loadHTML('<?xml ' . $xml . '>' . $code);
 
-			// remove the injected XML declaration again
-			$pis = $this->query('//processing-instruction()');
 
-			foreach (iterator_to_array($pis, false) as $pi) {
-				if ($pi->data === $xml) {
-					static::remove($pi);
+			// Newer libxml2 versions may not attach the injected XML node
+			// inside <html>. Instead, they may convert it into a top-level
+			// comment node that sits before <html>:
+			//   <!--?xml encoding="UTF-8" id="XYZ"--><html>...
+			//
+			//  XPath queries like //comment() or //processing-instruction()
+			//  often operate relative to the document element (<html>) and
+			//  therefore miss this node entirely.
+			//  To fix this, we must also inspect and clean up the document’s
+			//  top-level child nodes explicitly.
+			//
+			// Walk all top-level nodes of the document and remove
+			// any node that matches the injected XML marker
+			for ($node = $this->doc->firstChild; $node !== null; $node = $next) {
+				$next = $node->nextSibling;
+
+				if (
+					// Case 1: libxml preserved it as a processing instruction
+					($node->nodeType === XML_PI_NODE && $node->data === $xml) ||
+					// Case 2: libxml converted it into a comment node
+					// (<!--?xml encoding="UTF-8" id="..."-->)
+					($node->nodeType === XML_COMMENT_NODE && strpos($node->data, $xml) !== false)
+				) {
+					static::remove($node);
+					continue;
```

Comment:
`Question`: Shouldn't this be:


```suggestion
					break;
```